### PR TITLE
columns with degenerecies

### PR DESCRIPTION
**Pull request by ecoon**
This expands the applicability of columnar partitioning from no degenerecies are valid to degenerate edges but not cells are valid.

It is doable now to expand this further to degenerate cells will work, but I haven't done or tested that yet.  Will do so when I can find time.  In the meantime, this has been tested.

### CODE REVIEW

File: include/MSTK_private.h [hunk 1]
```diff
@@ -322,6 +322,7 @@ typedef enum MDelType {MDELREGION=-40, MDELFACE=-30, MDELEDGE=-20, MDELVERTEX=-1
   int        FixColumnPartitions_UpDown(Mesh_ptr mesh, MRegion_ptr mr, MFace_ptr* up, MFace_ptr* dn);
 
   int        MESH_Partition(Mesh_ptr parentmesh, int num, int *part, Mesh_ptr *submeshes);
+  int        MESH_Partition_Some(Mesh_ptr parentmesh, int num, int *part, int begin, int end, Mesh_ptr *submeshes);
```
Comment (raovgarimella) [inline include/MSTK_private.h:325]: I would rename this to be MESH_Partition_Partial

File: src/par/MESH_Partition.c [hunk 2]
```diff
@@ -526,6 +612,231 @@ extern "C" {
...
+          }
+
+          if (!found_lmf) {
+            lmf = MF_New(submeshes[part_index]);
+            MF_Set_GEntID(lmf,MF_GEntID(gmf));
+            MF_Set_GEntDim(lmf,MF_GEntDim(gmf));
+            mpart_no = MF_MasterParID(gmf);
+            if (mpart_no == 0) { /* not set */
+              MF_Set_MasterParID(lmf,part_no);
+              MF_Set_MasterParID(gmf,part_no+1);
```
Comment (raovgarimella) [inline src/par/MESH_Partition.c:684]: I think for what is being done in this PR, the master partition ID assignment will work as is. But as we are making changes to how we do partitioning and sending we may want to move the assignment of master partition ID on global mesh entities to a global step before we start making and sending submeshes off to different processors. Not sure why I did not do this before?

**Review by raovgarimella — CHANGES_REQUESTED** [review]
I believe this will work. However, there are many moving parts in this process and even though I tried to minimize the interdependence of the various steps, I am not sure nothing will break if changed. So, apart from the minor changes, I would strongly suggest a unit test or two to test this out. 

Also, I would also request you to separate the two things being fixed here - handling of column degeneracies and minimizing the memory usage in mesh partitioning.

**Comment by ecoon** [comment]
Wait, those should not be together? Please don't take the pull request if
you see mesh partition changes. I pushed those changes after making the
pull request, thinking that they wouldn't show up in the request. They
don't work!

I'll revert back and make sure you have a pull request for just the fix
columns stuff.

**Comment by certik** [comment]
I think this can be closed.

**Comment by raovgarimella** [comment]
These changes have been superseded by others

**Pull request closed without merging**